### PR TITLE
fix(fetch) always make sure that abort tracker is cleaned, revise ref count

### PR DESCRIPTION
**Pull request by cirospaciari**
### What does this PR do?
Make fetch more reliable and handle more edge cases, decrease the number of ref/deref
Fix: stream.blob() and similars to reject instead of throw
Fix: cookies should use `;` separator instead of `,` 

- [ ] Documentation or TypeScript types (it's okay to leave the rest blank in this case)
- [X] Code changes

### How did you verify your code works?
Existing tests + new tests

### CODE REVIEW

**Comment by robobun** [comment]
@cirospaciari, your commit 1029052e1abb3a92268424dc398651744adadfb0 is building: [`#2162`](https://buildkite.com/bun/bun/builds/2162)

File: src/bun.js/webcore/response.zig [hunk 1]
```diff
@@ -973,19 +976,15 @@ pub const Fetch = struct {
...
                     this.scheduled_response_buffer.reset();
                 }
-
-                this.mutex.unlock();
-                if (is_done) {
-                    const vm = globalThis.bunVM();
-                    this.poll_ref.unref(vm);
-                    this.deref();
+                if (any_js) {
+                    this.javascript_vm.drainMicrotasks();
```
Comment (Jarred-Sumner) [inline src/bun.js/webcore/response.zig]: ✂️ 
```suggestion

```

File: src/bun.js/webcore/response.zig [hunk 2]
```diff
@@ -1176,24 +1185,20 @@ pub const Fetch = struct {
 
                     promise_value.ensureStillAlive();
 
+                    // we need to drain the microtasks
+                    const loop = vm.eventLoop();
+                    loop.enter();
```
Comment (Jarred-Sumner) [inline src/bun.js/webcore/response.zig]: ✂️

File: src/bun.js/webcore/response.zig [hunk 3]
```diff
@@ -1230,26 +1230,40 @@ pub const Fetch = struct {
...
-                    var prom = self.promise.swap().asAnyPromise().?;
+                    // cleanup
+                    defer bun.default_allocator.destroy(self);
+                    defer self.held.deinit();
+                    defer self.promise.deinit();
+
+                    // we need to drain the microtasks
                     const globalObject = self.globalObject;
+                    const loop = globalObject.bunVM().eventLoop();
+                    loop.enter();
```
Comment (Jarred-Sumner) [inline src/bun.js/webcore/response.zig]: ✂️

File: packages/bun-usockets/src/context.c [hunk 4]
```diff
@@ -24,7 +24,7 @@
 #include <arpa/inet.h>
 #endif
 
-#define CONCURRENT_CONNECTIONS 6
+#define CONCURRENT_CONNECTIONS 4
```
Comment (Jarred-Sumner) [inline packages/bun-usockets/src/context.c:27]: nice

File: src/bun.js/webcore/response.zig [hunk 5]
```diff
@@ -1120,44 +1115,47 @@ pub const Fetch = struct {
...
+            }
 
+            const globalThis = this.global_this;
+            defer {
+                this.mutex.unlock();
+                // if we are not done we wait until the next call
+                if (is_done) {
+                    var poll_ref = this.poll_ref;
+                    this.poll_ref = .{};
+                    poll_ref.unref(vm);
```
Comment (Jarred-Sumner) [inline src/bun.js/webcore/response.zig:1139]: Is this on the main thread?

Comment (cirospaciari) [reply src/bun.js/webcore/response.zig:1139]: yes always

**Pull request merged**